### PR TITLE
Enhance Mellanox reboot cause test case

### PR DESCRIPTION
**Pull request by echuawu**
Add new reboot cause script for additional bios cpu and asic causes

### Description of PR

Summary:
Add new script to cover following reboot cause scenarios:

BIOS - In case the BIOS upgrade process ended with failure and cause the switch to reset.
CPU - Reset is initiated by SW on the CPU. it could be that SW encountered some catastrophic situation like a memory leak, eventually, the kernel reset the whole switch.
Reset from ASIC - Reset which is caused by ASIC.

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
Add test for https://github.com/sonic-net/sonic-platform-common/pull/277
#### How did you verify/test it?
Add test script for enhance reboot cause

#### Any platform specific information?
Mellanox platforms, except for SPC1 and SIMX

#### Supported testbed topology if it's a new test case?
Any topology

### CODE REVIEW

**Review by nhe-NV — DISMISSED** [review]

**Review by keboliu — APPROVED** [review]

File: tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py [hunk 1]
```diff
@@ -138,7 +147,7 @@ def __init__(self, dut):
         :param dut: DUT object representing a SONiC switch under test.
         """
         self.dut = dut
-        #self.unlink_file_list = {}
+        # self.unlink_file_list = {}
```
Comment (liat-grozovik) [inline tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py:150]: can you clean this code?

File: tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py [hunk 2]
```diff
@@ -913,6 +923,12 @@ def mock_data(self):
         platform_data = get_platform_data(self.mock_helper.dut)
         thermal_dict = platform_data["thermals"]
         for category, content in thermal_dict.items():
+            # TODO: temp local solution, since the "sodimm" is only supported in master,
```
Comment (liat-grozovik) [inline tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py:926]: why not checking on branch instead of todo?

**Review by liat-grozovik — COMMENTED** [review]
LGTM see minor comments

File: tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py (hunk 1 above)
Comment (echuawu) [reply tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py:150]: This modification was the update of pre-commit check. My code update does not contain this part, but it failed at pre-commit process.

File: tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py (hunk 2 above)
Comment (echuawu) [reply tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py:926]: This is also the original code which I did not modified. It was removed for some reason when I am handling the updates according to pre-commit check result in the seconds commit. I readd it again.

File: tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py (hunk 1 above)
Comment (echuawu) [reply tests/platform_tests/mellanox/mellanox_thermal_control_test_helper.py:150]: All the codes I need to update was in the first commit. And other code changes are all for passing through the pre-commit check. There are many old codes were not checked by pre-commit then, but when I modified some parts of the old code, pre-commit would check the old codes too, so I edit them in order to pass the check.

**Comment by echuawu** [comment]
This PR is dropped and reopens a new one: https://github.com/sonic-net/sonic-mgmt/pull/6944

**Pull request closed without merging**